src/HealthGPS/runner.cpp

```diff
@@ -107,27 +95,23 @@ double ModelRunner::run(Simulation &baseline, Simulation &intervention,
         }
     }
 
-    /* Terminate simulation */
-    baseline.terminate();
-    intervention.terminate();
-
     ElapsedTime elapsed = std::chrono::steady_clock::now() - start;
     auto elapsed_ms = elapsed.count();
 
     notify(std::make_unique<RunnerEventMessage>(runner_id_, RunnerAction::finish, elapsed_ms));
     return elapsed_ms;
 }
 
-bool ModelRunner::is_running() const noexcept { return running_.load(); }
+bool Runner::is_running() const noexcept { return running_.load(); }
 
-void ModelRunner::cancel() noexcept {
+void Runner::cancel() noexcept {
     if (is_running()) {
         source_.request_stop();
     }
 }
 
-void ModelRunner::run_model_thread(const std::stop_token &token, Simulation &model,
-                                   unsigned int run, const std::optional<unsigned int> seed) {
+void Runner::run_model_thread(const std::stop_token &token, Simulation &model, unsigned int run,
+                              const unsigned int seed) {
```

Comment:
I have a vague feeling we changed `seed` to be an optional type at some point, because previously the code treated `seed==0` as meaning "don't seed the RNG" and using an optional seemed cleaner.

Reply:
I had that feeling also, but have checked the code and a seed is set for *all* simulations in `Runner::run` regardless. Maybe you were thinking of the RNG 'seed generator' RNG? The one in `program.cpp:137`, that generates seeds for job RNGs?

It's a bit confusing, hence the changes.